storage/grpc_client.go

```diff
@@ -162,6 +162,8 @@ func (c *grpcStorageClient) CreateBucket(ctx context.Context, project, bucket st
 		b.Location = "US"
 	}
 
+	// TO-DO: implement ObjectRetention once available - see b/308194853
```

Comment:
I think for now we should return a NotImplemented error if a user tries to pass this, so that it doesn't just silently not happen.

Reply:
I meant the gRPC unimplemented error code: see https://pkg.go.dev/google.golang.org/grpc@v1.60.0/codes#Unimplemented . I think we can just make and return this.